astronomy/gravity_models.cfg

```diff
@@ -0,0 +1,218 @@
+principia_gravity_models {
+  // Axis directions from "Report of the IAU Working Group on Cartographic
+  // Coordinates and Rotational Elements: 2009", Archinal et al.,
+  // http://astropedia.astrogeology.usgs.gov/download/Docs/WGCCRE/WGCCRE2009reprint.pdf.
+  // The values given are for J2000.
+  // TODO(egg): actually compute the value at J2000, instead of using the
+  // constant term; some of the higher-order terms don't vanish at J2000.0.
+
+  // Unless mentioned otherwise, gravitational parameters are from the SPICE
+  // kernel for DE431,
+  // http://naif.jpl.nasa.gov/pub/naif/generic_kernels/pck/gm_de431.tpc.
+
+  // Some sources for gravity models give C2,0 instead of J2.
+  // In that case, J2 was computed using Mathematica, as
+  //   N[SetPrecision[-c20, Infinity] * Sqrt[5], 17].
+
+  // Star.
+  body {
+    // J2 coefficient and reference radius from
+    // http://ilrs.gsfc.nasa.gov/docs/2014/196C.pdf.
+    name                    = Sun
+    gravitational_parameter = 1.3271244004193938E+11 km^3/s^2
+    axis_right_ascension    = 286.13 deg
+    axis_declination        = 63.87 deg
+    j2                      = 2.1106088532726840E-07
```

Comment:
Preference for having all the `j2` in units of `1E-6`, it makes it easier to compare them.